runtime/flangrti/CMakeLists.txt

```diff
@@ -96,14 +96,14 @@ if (NOT DEFINED LIBOMP_EXPORT_DIR)
     FLANG_LIBOMP
     NAMES omp libomp
     HINTS ${CMAKE_BINARY_DIR}/lib)
-  target_link_libraries(flangrti_shared PRIVATE ${FLANG_LIBOMP})
+  target_link_libraries(flangrti_shared PUBLIC ${FLANG_LIBOMP})
```

Comment:
this commit deserves separate PR

Reply:
All 10 commits are independent and they all require separate PRs? I'd rather have one PR than 10 PRs which would take forever to review because of the lack of reviewers.

Reply:
Changing the link dependencies of libraries has a much wider impact than the rest of the changes in this PR. Why is this change necessary?

Reply:
It's described in the commit message.

>  libomp and libpgmath are dependencies of libflangrti's downstream libflang

> Since flang runtime library libflang has fortran files and they depend on
libomp and libpgmath, they should be public